### DISCUSSION TITLE
Add support for more than just SSH connections (docker, kvm, ssh, local install, etc...)

### DISCUSSION THREAD

**Discussion by austinkregel** [vitodeploy/vito#230] [Ideas]
As I've been going over the code, I don't see a technical limitation preventing us from supporting more than just SSH connections 🤔 

Partially related to https://github.com/vitodeploy/vito/issues/22, I couldn't get the linked feature website to load and I didn't want the request to get lost

**Comment by saeedvaziry** [comment]
@austinkregel you mean to provision docker containers? or you mean to install VitoDeploy itself via docker?

**Reply by austinkregel** [reply]
I mean to provision docker containers, virtual machines, or localhost (the server VitoDeploy is on).

I'm not seeing a technical limitation to installing vhosts on the Vito server 👀 

Adding support for them could get us one step closer to having Vito manage a Vito installation 👀

**Reply by austinkregel** [reply]
I know there's been a variety of discussions around a similar kind of configuration where Vito has other vhosts on the Vito server like this in this thread https://github.com/vitodeploy/vito/discussions/158

I understand it _could_ be potentially dangerous, and think a good implementation would ensure the user is warned before any potentially dangerous action is taken, and there are safeguards in places to prevent breaking the installation.

Using `ngnix reload` for example instead of restarting the server could enable us to detect nginx config issues 👀 I've got some ideas around how we could handle this 🤔

**Comment by saeedvaziry** [comment]
> I mean to provision docker containers, virtual machines, or localhost (the server VitoDeploy is on).

TBH I am still not getting your point of docker containers :), Vito is built to make it easy to provision the virtual servers but for docker you just need a docker-compose file. Also there are a lot of tools provisioning docker which doesn't really make sense to reinvent the wheel here.

We can still make it possible to deploy to the same server (VitoDeploy) but I believe it will be really messed up and I wouldn't take the risk of putting the server manager of your server inside the server itself :)

> Using ngnix reload for example instead of restarting the server could enable us to detect nginx config issues 👀 I've got some ideas around how we could handle this 🤔

Clearly we can do some stuff here, since now we support exit codes, it would make more sense to handle such errors. and of course PRs are welcome. Feel free to also add it to the [roadmap](https://vitodeploy.featurebase.app/).
